packages/simcore-sdk/src/simcore_sdk/node_ports_v2/nodeports_v2.py

```diff
@@ -28,6 +30,28 @@
 log = logging.getLogger(__name__)
 
 
+#  -> @GitHK this looks very dangerous, using a lot of protected stuff, just checking the number of ignores shows it's a bad idea...
```

Comment:
Maybe a test could be added so is caught in case something goes wrong?

Reply:
Please be my guest. that is your code @GitHK and these 2 functions were added by you. I did not try to understand too much of them, they look complex enough.